universe/Planet.cpp

```diff
@@ -481,50 +481,69 @@ float Planet::NextTurnCurrentMeterValue(MeterType type) const {
 
 std::string Planet::CardinalSuffix() const {
     std::string retval = "";
+    if (ID() == INVALID_OBJECT_ID) {
+        WarnLogger() << "Planet " << Name() << " has invalid ID";
+        return retval;
+    }
+
     // Planets are grouped into asteroids, and non-asteroids
     // Asteroids receive a localized prefix
     if (Type() == PT_ASTEROIDS)
-        retval.append(UserString("NEW_ASTEROIDS_SUFFIX") + " ");
+        retval.append(UserString("NEW_ASTEROIDS_SUFFIX"));
 
     std::shared_ptr<System> cur_system = GetSystem(SystemID());
     if (cur_system) {
         if (cur_system->OrbitOfPlanet(ID()) < 0) {
             ErrorLogger() << "Planet " << Name() << "(" << ID() << ") "
                           << "has no current orbit";
+            if (Type() == PT_ASTEROIDS)
```

Comment:
The asteroids had a prefix applied and needed a space to separate the ordinal.
Determining to add a comment or change the functionality, I opted for later which cascaded into improving the flow of this function.